Source/WebGPU/WGSL/TypeCheck.cpp

```diff
@@ -205,9 +222,25 @@ void TypeChecker::visit(AST::Expression&)
 
 void TypeChecker::visit(AST::FieldAccessExpression& access)
 {
-    auto* structType = infer(access.base());
-    // FIXME: implement member lookup once we have a struct type
-    inferred(structType);
+    auto* baseType = infer(access.base());
+    if (isBottom(baseType)) {
+        inferred(m_types.bottomType());
+        return;
+    }
+
+    if (std::holds_alternative<Struct>(*baseType)) {
+        auto& structType = std::get<Struct>(*baseType);
```

Comment:
nit: This feels like the pattern in `WGSL::AST` that is written:
```
    if (is<Struct>(*baseType)) {
        auto& structType = downcast<Struct>(*baseType);
```

I would like to err on the side of consistency in the compiler. (Also, after looking at built times with `-ftime-trace`, I hate how slow `std::variant`s are to compile)